src/server/main_service.h

```diff
@@ -134,9 +134,13 @@ class Service : public facade::ServiceInterface {
 
   void Discard(CmdArgList args, Transaction* tx, SinkReplyBuilder* builder,
                ConnectionContext* cntx);
-  void Eval(CmdArgList args, Transaction* tx, SinkReplyBuilder* builder, ConnectionContext* cntx);
-  void EvalSha(CmdArgList args, Transaction* tx, SinkReplyBuilder* builder,
-               ConnectionContext* cntx);
+  void Eval(CmdArgList args, Transaction* tx, SinkReplyBuilder* builder, ConnectionContext* cntx,
+            bool read_only = false);
```

Comment:
I think it'd be slightly more elegant to keep `Eval` with no `read_only` default argument, and add an internal (`.cc`-only) function, say `EvalInternal` which accepts the read-only bit.
It's not too important, so I'll leave the decision to you. If you do decide to follow up on this, consider making that bit an `enum` (like `enum class EvalMode { kReadOnly, kReadWrite }`) and pass that as the argument. This makes the call site more readable.

Reply:
@chakaz you love Enums :rofl: I am saying this because every time there is a bool as parameter a switch flips within you :rofl:

Reply:
Only for function arguments :nerd_face:

Reply:
Sure will update

Reply:
I think this is perfectltly fine. We list the family commands as members for no valid reason either 🙂